runtime/src/main/java/org/corfudb/runtime/object/transactions/OptimisticTransactionalContext.java

```diff
@@ -236,14 +246,20 @@ public long commitTransaction() throws TransactionAbortedException {
             return commitAddress;
         }
 
+        // Write to the transaction stream if transaction logging is enabled
+        Set<UUID> affectedStreams = new HashSet<>(writeSet.keySet());
+        if (this.builder.runtime.getObjectsView().isTransactionLogging()) {
+            affectedStreams.add(TRANSACTION_STREAM_ID);
+        }
+
         // Now we obtain a conditional address from the sequencer.
         // This step currently happens all at once, and we get an
         // address of -1L if it is rejected.
         long address = this.builder.runtime.getStreamsView()
                 .acquireAndWrite(
 
                         // a set of stream-IDs that contains the affected streams
-                        writeSet.keySet(),
+                        affectedStreams,
```

Comment:
I think that the same thing needs to be done to WriteAfterWriteTransactionContext